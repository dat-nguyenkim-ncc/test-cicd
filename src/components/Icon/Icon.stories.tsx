import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import Icon, { IconProps } from './Icon'
import Path from './Paths'

export default {
  title: 'Primitives/Icon',
  component: Icon,
  parameters: {
    icon: {
      control: {
        type: 'inline-radio',
        values: Object.keys(Path),
      },
    },
  },
} as Meta

const Template: Story<IconProps> = args => <Icon {...args} />

export const Default = Template.bind({})
Default.args = {
  icon: 'download',
} as IconProps
