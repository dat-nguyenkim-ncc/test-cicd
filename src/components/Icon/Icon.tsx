import React from 'react'
import { Flex } from 'theme-ui'
import { MapVariantToSize, Size, ViewInterface } from '../../types'
import { PaletteKeys } from '../../theme'
import Path, { Paths } from './Paths'

const sizes: MapVariantToSize = {
  big: 80,
  small: 24,
  tiny: 16,
  normal: 40,
}

export type IconProps = {
  color?: PaletteKeys
  icon: Paths | null
  background?: PaletteKeys
  isSquare?: boolean
  size?: Size
  iconSize?: number
}

type Props = ViewInterface<IconProps>

const Icon = ({ background, icon, sx, color = 'text', isSquare, size, iconSize }: Props) => {
  const IconPath = icon ? Path[icon] : null

  return (
    <Flex
      as="span"
      sx={{
        backgroundColor: background || 'transparent',
        borderRadius: !isSquare ? '100%' : 'inherit',
        width: background && size ? sizes[size] : 'auto',
        height: background && size ? sizes[size] : 'auto',
        justifyContent: 'center',
        alignItems: 'center',
        ...sx,
      }}
    >
      {IconPath && (
        <svg width={iconSize || 16} height={iconSize || 16} viewBox="0 0 16 16" fill="none">
          <IconPath color={color} />
        </svg>
      )}
    </Flex>
  )
}

export default Icon
