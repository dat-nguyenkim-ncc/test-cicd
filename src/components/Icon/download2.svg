<svg width="14" height="15" viewBox="0 0 14 15" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M10.5 6.76091C10.4291 6.6059 10.2734 6.50717 10.1019 6.50717H8.35187V0.878303C8.35187 0.639292 8.15587 0.445312 7.91437 0.445312H6.16438C5.92288 0.445312 5.72688 0.639292 5.72688 0.878303V6.50717H3.97687C3.80537 6.50717 3.64963 6.60676 3.57875 6.76091C3.507 6.91592 3.535 7.09691 3.64788 7.22507L6.71038 10.689C6.7935 10.7834 6.91337 10.8371 7.03937 10.8371C7.16537 10.8371 7.28525 10.7825 7.36838 10.689L10.4309 7.22507C10.5446 7.09777 10.5709 6.91592 10.5 6.76091Z" fill="currentColor"/>
<path d="M12.1333 9.82617V12.5973H1.86667V9.82617H0V13.521C0 14.0318 0.418133 14.4447 0.933333 14.4447H13.0667C13.5828 14.4447 14 14.0318 14 13.521V9.82617H12.1333Z" fill="currentColor"/>
</svg>
