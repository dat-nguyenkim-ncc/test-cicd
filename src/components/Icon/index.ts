export { default } from './Icon'
export type { Paths } from './Paths'
export type { IconProps } from './Icon'
