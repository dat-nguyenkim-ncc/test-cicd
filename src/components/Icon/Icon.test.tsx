import * as React from 'react'
import { shallow } from 'enzyme'
import Icon from './Icon'

describe('Icon', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<Icon icon="download" />)
    expect(wrapper).toMatchSnapshot()
  })
})
