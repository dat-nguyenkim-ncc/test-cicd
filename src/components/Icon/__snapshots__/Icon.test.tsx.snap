// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon matches snapshot 1`] = `ShallowWrapper {}`;
